pennylane/gradients/general_shift_rules.py

```diff
@@ -396,8 +396,12 @@ def _copy_and_shift_params(tape, indices, shifts, multipliers, cast=False):
 
         # Shift copied parameter
         new_params = list(op.data)
-        new_params[p_idx] = new_params[p_idx] * qml.math.convert_like(multiplier, new_params[p_idx])
-        new_params[p_idx] = new_params[p_idx] + qml.math.convert_like(shift, new_params[p_idx])
+        multiplier = qml.math.convert_like(multiplier, new_params[p_idx])

```

Comment:
We are now passing interface-specific variables to the gradients module.  Without the `cast_like`, Tensorflow was raising errors.